### ISSUE TITLE
Feature Request: Be able to provide custom `yamux.Config` to `NewRPCClient`

### ISSUE THREAD

**Issue by mickmister** [hashicorp/go-plugin#248]
In order to use a debugger while developing a plugin, I need to modify the below code to allow for the RPC connection's timeout to tolerate the amount of time the debugger is paused. Before making this change, the plugin gets disconnected due to the failing timeout.

https://github.com/hashicorp/go-plugin/blob/e889c1ba1044d1b183d4ed45d0f2f531db3ba50b/rpc_client.go#L64-L66

```go
func NewRPCClient(conn io.ReadWriteCloser, plugins map[string]Plugin) (*RPCClient, error) { 
    sessionConfig := yamux.DefaultConfig()
    sessionConfig.EnableKeepAlive = true
    sessionConfig.ConnectionWriteTimeout = time.Minute * 5

    // Create the yamux client so we can multiplex 
    mux, err := yamux.Client(conn, sessionConfig)
```

I'd like to be able to provide a value to this function, in order to allow for customizing the `yamux.Config` used here. I'd like to make the change to make this configurable, possibly by environment variables if that's the most appropriate way. Any advice on how this change should be done?

**Comment by mickmister** [comment]
Does anyone mind if I try to implement this?

**Comment by b-dean** [comment]
@mickmister, is this a duplicate of #205?

also maybe you can use porter's fork

```go.mod
// go.mod
replace github.com/hashicorp/go-plugin => github.com/getporter/go-plugin v1.4.4-porter.1
```

in that `NewRPCClient` takes a `*yamux.Config`: https://github.com/getporter/go-plugin/blob/v1.4.4-porter.1/rpc_client.go#L64

**Comment by mickmister** [comment]
> @mickmister, is this a duplicate of https://github.com/hashicorp/go-plugin/issues/205?

@b-dean Yes it seems I made both of the issues 🤦  Closing in favor of https://github.com/hashicorp/go-plugin/issues/205
